api/backend/ecs/task_manager.go

```diff
@@ -50,88 +52,66 @@ func (this *ECSTaskManager) ListTasks() ([]string, error) {
-	ecsEnvironmentID := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
-	ecsTaskID := id.L0TaskID(taskID).ECSTaskID()
+func (this *ECSTaskManager) GetTask(environmentID, taskARN string) (*models.Task, error) {
+	clusterName := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
+	task, err := this.ECS.DescribeTask(clusterName.String(), taskARN)
+	if err != nil {
+		return nil, err
+	}
+
+	return modelFromTasks([]*ecs.Task{task})
+}
 
-	tasks, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(ecsTaskID.String()))
+func (this *ECSTaskManager) GetEnvironmentTasks(environmentID string) (map[string]*models.Task, error) {
+	clusterName := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
+	taskDescriptions, err := this.ECS.DescribeEnvironmentTasks(clusterName.String(), id.PREFIX)
 	if err != nil {
 		return nil, err
 	}
 
-	taskDescs := []*ecs.Task{}
-	if len(tasks) > 0 {
-		taskDescs, err = this.describeTasks(ecsEnvironmentID, tasks)
+	taskARNModels := map[string]*models.Task{}
+	for _, taskDescription := range taskDescriptions {
+		task, err := modelFromTasks([]*ecs.Task{taskDescription})
 		if err != nil {
 			return nil, err
 		}
+
+		taskARNModels[aws.StringValue(taskDescription.TaskArn)] = task
 	}
 
-	return modelFromTasks(taskDescs)
+	return taskARNModels, nil
 }
 
-func (this *ECSTaskManager) DeleteTask(environmentID, taskID string) error {
+func (this *ECSTaskManager) DeleteTask(environmentID, taskARN string) error {
 	ecsEnvironmentID := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
-	ecsTaskID := id.L0TaskID(taskID).ECSTaskID()
-
-	taskARNs, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(ecsTaskID.String()))
-	if err != nil {
-		return err
-	}
-
-	// This stops the task, later reaping by AWS will prevent it from being returned.
-	reason := "Task stopped by User"
-
-	for _, taskARN := range taskARNs {
-		if err := this.ECS.StopTask(ecsEnvironmentID.String(), reason, *taskARN); err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return this.ECS.StopTask(ecsEnvironmentID.String(), taskARN, StopTaskReason)
 }
 
 func (this *ECSTaskManager) CreateTask(
 	environmentID string,
-	taskName string,
 	deployID string,
 	overrides []models.ContainerOverride,
-) (*models.Task, error) {
+) (string, error) {
 	ecsEnvironmentID := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
 	ecsDeployID := id.L0DeployID(deployID).ECSDeployID()
 
-	//taskID := id.GenerateHashedEntityID(taskName)
-	//ecsTaskID := id.L0TaskID(taskID).ECSTaskID()
-
 	ecsOverrides := []*ecs.ContainerOverride{}
 	for _, override := range overrides {
 		o := ecs.NewContainerOverride(override.ContainerName, override.EnvironmentOverrides)
 		ecsOverrides = append(ecsOverrides, o)
 	}
 
 	startedBy := id.PREFIX
-	tasks, failed, err := this.ECS.RunTask(ecsEnvironmentID.String(), ecsDeployID.TaskDefinition(), 1, stringp(startedBy), ecsOverrides)
+	task, err := this.ECS.RunTask(ecsEnvironmentID.String(), ecsDeployID.TaskDefinition(), startedBy, ecsOverrides)
```

Comment:
was this an SDK change that `failed` is no longer returned?

Reply:
no, we changed our aws layer to make this call more user-friendly.